components/resourceDrawer/index.jsx

```diff
@@ -42,39 +44,42 @@ export function handleCloseDrawer({ setMenuOpen, router }) {
 export default function ResourceDrawer({ onUpdate }) {
   const { menuOpen, setMenuOpen } = useContext(DetailsMenuContext);
   const { fetch } = useSession();
-  const [datasetWithAcl, setDatasetWithAcl] = useState(null);
-  const [loading, setLoading] = useState(true);
-
+  const router = useRouter();
   const {
     query: { action, resourceIri },
-  } = useRouter();
-
-  const router = useRouter();
+  } = router;
+  const encodedResourceIri = encodeURI(resourceIri);
+  const { data: resourceInfo, error: resourceError } = useResourceInfo(
+    encodedResourceIri
+  );
+  const { accessControl, error: accessControlError } = useAccessControl(
+    resourceInfo,
+    fetch
+  );
 
   useEffect(() => {
     setMenuOpen(!!(action && resourceIri));
   }, [action, resourceIri, setMenuOpen]);
 
   useEffect(() => {
-    if (!resourceIri) return;
-    setLoading(true);
-    const encodedResourceIri = encodeURI(resourceIri);
-    getResourceInfoWithAcl(encodedResourceIri, { fetch }).then((dataset) => {
-      setDatasetWithAcl(dataset);
-      setLoading(false);
-    });
-  }, [fetch, resourceIri]);
+    if (accessControlError || resourceError) {
+      throw accessControlError || resourceError;
+    }
+  });
```

Comment:
does this need to be in a useEffect anymore?

Reply:
Fair point, it doesn't need to no. We should also handle the error better. I'll create a ticket to handle this, since the error handling can be done in various ways.